app/workers/education_form/delete_old_applications.rb

```diff
@@ -4,9 +4,32 @@ class DeleteOldApplications
     include Sidekiq::Worker
 
     def perform
-      records = EducationBenefitsClaim.where("processed_at < '#{2.months.ago}'")
-      logger.info("Deleting #{records.count} old 22-1990s")
-      records.delete_all
+      edu_claim_ids = []
+      saved_claim_ids = []
+
+      # Remove old education benefits claims and saved claims older than 2 months
+      EducationBenefitsClaim.eager_load(:saved_claim)
+                            .where("processed_at < '#{2.months.ago}'")
+                            .find_each do |record|
+        edu_claim_ids << record.id
+        saved_claim_ids << record&.saved_claim.id
+      end
+
+      # Remove saved claims with no matching education benefits claim
```

Comment:
i think this should be moved to a rake task. once this job is deleting saved claims along with edu claims then we should be able to run the task once and never have any more orphaned saved claims